group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2553,6 +2555,10 @@ private CoordinatorResult<Void, Record> expireClassicGroupMemberHeartbeat(
             log.info("Received notification of heartbeat expiration for member {} after group {} " +
                     "had already been unloaded or deleted.",
                 memberId, group.groupId());
+        } else if (!containsClassicGroup(group.groupId())) {
+            log.info("Received notification of heartbeat expiration for member {} after group {} " +
+                    "had already been deleted or upgraded.",
+                memberId, group.groupId());
```

Comment:
I have a similar comment for this one. However, in this case, we could just replace the `group` argument by `groupId` and do the lookup here.